ts/features/payments/checkout/screens/WalletPaymentPickMethodScreen.tsx

```diff
@@ -60,13 +64,67 @@ const WalletPaymentPickMethodScreen = () => {
   const [waitingTransactionActivation, setWaitingTransactionActivation] =
     React.useState(false);
 
+  const userPaymentMethods = React.useMemo(
+    () =>
+      pipe(
+        userWalletsPots,
+        pot.toOption,
+        O.getOrElse(() => [] as Wallets["wallets"])
+      ),
+    [userWalletsPots]
+  );
+
+  const allPaymentMethods = React.useMemo(
+    () =>
+      pipe(
+        paymentMethodsPot,
+        pot.toOption,
+        O.getOrElse(() => [] as PaymentMethodsResponse["paymentMethods"])
+      ),
+    [paymentMethodsPot]
+  );
+
   useFocusEffect(
     React.useCallback(() => {
       dispatch(paymentsGetPaymentMethodsAction.request());
       dispatch(paymentsGetPaymentUserMethodsAction.request());
     }, [dispatch])
   );
 
+  // If the user doesn't have any onboarded payment method and the backend doesn't return any payment method as guest ..
+  // .. we redirect the user to the outcome screen with an outcome that allow the user to start the onboarding process of a new payment method.
+  // .. This implementation will be removed as soon as the backend will migrate totally to the NPG. (https://pagopa.atlassian.net/browse/IOBP-632)
+  useEffect(() => {
+    if (
+      pot.isSome(paymentMethodsPot) &&
+      _.isEmpty(allPaymentMethods) &&
+      pot.isSome(userWalletsPots) &&
+      _.isEmpty(userPaymentMethods) &&
+      pot.isSome(paymentDetailsPot)
```

Comment:
Can we move all this logic in a selector? For example: `hasValidPaymentMethodsSelector`, which returns `true` or `false` following the logic above.

Reply:
Good idea! Addressed into 20ea0b98f1eab6a7677c8ebba823be0c83ba58b5